pkg/server/endpoints/auth.go

```diff
@@ -0,0 +1,170 @@
+package endpoints
+
+import (
+	"context"
+	"crypto"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/andres-erbsen/clock"
+	"github.com/spiffe/go-spiffe/v2/bundle/x509bundle"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
+	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
+	"github.com/spiffe/spire/pkg/common/telemetry"
+	"github.com/spiffe/spire/pkg/common/x509util"
+	"github.com/spiffe/spire/pkg/server/cache/dscache"
+	"github.com/spiffe/spire/proto/spire/common"
+)
+
+var (
+	lastMisconfigLogTime = new(atomic.Int64)
+	misconfigClk         = clock.New()
+)
+
+const misconfigLogEvery = time.Minute
+
+// shouldLogFederationMisconfiguration returns true if the last time a misconfiguration
+// was logged was more than misconfigLogEvery ago.
+func shouldLogFederationMisconfiguration() bool {
```

Comment:
hmm... should we be doing this logging rate limiting per trust domain?

Reply:
Yep, that's a good idea!